interfaces/builtin/gpio_chardev.go

```diff
@@ -0,0 +1,202 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/features"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/systemd"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+	"github.com/snapcore/snapd/strutil"
+)
+
+// TODO: Snapd should validate the correctness of slot declarations
+// (i.e. lines across slots are unique) when installing the gadget
+// snap e.g. when validating snap.yaml
+
+// https://docs.kernel.org/userspace-api/gpio/chardev.html
+// https://docs.kernel.org/admin-guide/gpio/gpio-aggregator.html
+const gpioChardevSummary = `allows access to specific GPIO chardev lines`
```

Comment:
We don't do that often/at all, but maybe you could add a TLDR summary of how the interface is expected to work:
```
The interface operates as follows:
- uses snap-gpio-helper to set up a virtual GPIO device exposing specific
  lines defined in the slot as character device node at 
  /dev/snap/gpio-chardev/<slot-snap>/<slot-name>
- sets up a symlink at /dev/snap/gpio-chardev/<plug-snap>/<plug-name>
  pointing at the virtual slot device
- slot devices are given a specific udev tag, which consumer rules match on
```

Reply:
we don't do it because summary is meant to be a one liner. We can leave a comment though, and we will need good docs for this

Reply:
sorry, I meant a TLDR summary in the comment, not the actual summary summary of the interface